### PR TITLE
[improvement](multi catalog)paimon support filesystem metastore

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

1.support filesystem metastore

2.support predicate and project when split

3.fix some bug

todo:

1.add ut
2.Now you need to manually put paimon-s3-0.4.0-incubating.jar in be/lib/java_extensions when use s3 filesystem

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.65 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          88.2 seconds inserted 10000000 Rows, about 113K ops/s
 storage size: 17167039728 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230707105739_clickbench_pr_174053.html

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.38 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          87.2 seconds inserted 10000000 Rows, about 114K ops/s
 storage size: 17164257560 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230711235711_clickbench_pr_176652.html

File: fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java [hunk 1]
```diff
@@ -18,97 +18,103 @@
...
     private RecordReader<InternalRow> reader;
     private final PaimonColumnValue columnValue = new PaimonColumnValue();
+    private List<String> paimonAllFieldNames;
 
     public PaimonJniScanner(int batchSize, Map<String, String> params) {
-        metastoreUris = params.get("hive.metastore.uris");
-        warehouse = params.get("warehouse");
-        splitAddress = Long.parseLong(params.get("split_byte"));
-        lengthByte = Integer.parseInt(params.get("length_byte"));
+        LOG.info("params:{}", params);
```
Comment (morningman) [inline fe/be-java-extensions/paimon-scanner/src/main/java/org/apache/doris/paimon/PaimonJniScanner.java:61]:
```suggestion
        LOG.debug("params:{}", params);
```

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/paimon/PaimonPredicateConverter.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import org.apache.paimon.types.DataField;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.types.RowType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+public class PaimonPredicateConverter {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/paimon/PaimonPredicateConverter.java:39]: Add some unit test for this converter

**Comment by morningman** [comment]
Also update the document please

**Pull request closed without merging**